src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx

```diff
@@ -0,0 +1,231 @@
+// (C) 2019-2020 GoodData Corporation
+import * as React from "react";
+import { render } from "react-dom";
+import { VisualizationObject, AFM } from "@gooddata/typings";
+
+import cloneDeep = require("lodash/cloneDeep");
+import get = require("lodash/get");
+import set = require("lodash/set");
+
+import * as BucketNames from "../../../../constants/bucketNames";
+import {
+    IReferencePoint,
+    IExtendedReferencePoint,
+    IVisConstruct,
+    IVisualizationProperties,
+    IVisProps,
+    IBucketItem,
+    IBucket,
+} from "../../../interfaces/Visualization";
+import { PluggableBaseChart } from "../baseChart/PluggableBaseChart";
+import { BUCKETS, METRIC, ATTRIBUTE } from "../../../constants/bucket";
+import { GEO_PUSHPIN_CHART_UICONFIG } from "../../../constants/uiConfig";
+import {
+    sanitizeFilters,
+    getBucketItemsByType,
+    getPreferredBucketItems,
+    hasBucket,
+} from "../../../utils/bucketHelper";
+import { setGeoPushpinUiConfig } from "../../../utils/uiConfigHelpers/geoPushpinChartUiConfigHelper";
+import { removeSort, createSorts } from "../../../utils/sort";
+import { getReferencePointWithSupportedProperties } from "../../../utils/propertiesHelper";
+import { VisualizationTypes } from "../../../../constants/visualizationTypes";
+
+import UnsupportedConfigurationPanel from "../../configurationPanels/UnsupportedConfigurationPanel";
+import { DASHBOARDS_ENVIRONMENT } from "../../../constants/properties";
+import { GeoChart } from "../../../../components/core/GeoChart";
+import { IGeoConfig } from "../../../../interfaces/GeoChart";
+
+export class PluggableGeoPushpinChart extends PluggableBaseChart {
+    private geoPushpinElement: string;
+
+    constructor(props: IVisConstruct) {
+        super(props);
+
+        const { callbacks, element, visualizationProperties } = props;
+        this.type = VisualizationTypes.PUSHPIN;
+        this.supportedPropertiesList = [];
+        this.callbacks = callbacks;
+        this.geoPushpinElement = element;
+        this.initializeProperties(visualizationProperties);
+    }
+
+    public getExtendedReferencePoint(referencePoint: IReferencePoint): Promise<IExtendedReferencePoint> {
+        const clonedReferencePoint = cloneDeep(referencePoint);
+        let newReferencePoint: IExtendedReferencePoint = {
+            ...clonedReferencePoint,
+            uiConfig: cloneDeep(GEO_PUSHPIN_CHART_UICONFIG),
+        };
+
+        const buckets = get(clonedReferencePoint, BUCKETS, []);
+        const locations = getBucketItemsByType(buckets, BucketNames.LOCATION, [ATTRIBUTE]);
+        const measuresSize = this.getMeasureSizeBucketItem(buckets);
+        const measuresColor = this.getMeasureColorBucketItem(buckets);
+        const segments = getPreferredBucketItems(
+            buckets,
+            [BucketNames.STACK, BucketNames.SEGMENT_BY],
```

Comment:
what about SEGMENT?

Reply:
We want to follow bucket name so we leave it as its name.
If we refactor it here so we have to update it in component interface as well
```
<GeoPushpinChart
...
    // segmentBy={segmentBy}
    segment={segment}
...
/>

Reply:
yes but why you prefere the STACK bucket and not "old" SEGMENT bucket from other charts? It is connected with bucket localIdentifiers discussion above. With every newly introduced bucket localIdentifier with semantic of segment/stack you will need to go and update this line of code.

Reply:
Fixed